nima_core/precognition.py

```
@@ -83,6 +83,25 @@ def __init__(
         self.lookback_days = lookback_days
         self._conn: Optional[sqlite3.Connection] = None
 
+    def close(self) -> None:
+        """Close the SQLite connection and release resources."""
+        if self._conn is not None:
+            try:
+                self._conn.close()
+            except Exception:
+                pass
```

Ruff (0.15.6)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)